db/version_set.cc

```diff
@@ -688,6 +689,12 @@ class FilePickerMultiGet {
                user_comparator_->CompareWithoutTimestamp(
```

Comment:
Overall I found the purpose of "if (cmp_largest == 0)" hard to understand and the comment does not help much. 

My main questions is:
If we agree with _"it keeps moving forward until the last key in the batch that falls in that file"_ as mentioned in https://github.com/facebook/rocksdb/pull/12295/files#diff-6270b3486fea620597e24151dd0d75c2c14b3c4c30d62e0567811723628733deR623-R625, I don't know why we can't merge the case cmp_largest == 0 with cmp_largest < 0. 

Particularly, the comment about this case "_which means the next key will not be in this file, so stop looking further_" seems to assume there aren't any duplicate keys. If we don't plan to change/clarify MultiGet() on that aspect, I don't know why we still keep this special path. Also, should we still allow MultiGet() to take duplicate keys for one CF? Duplicate keys makes the logic harder to reason.

For the above question, I was once wondering if it's because we need to, as the comment mentioned, _"..leave batch_iter_ as is since we may have to pick up from there for the next file, if this file has a merge value rather than"_. But again, I don't understand we didn't do the same ("leave batch_iter_ as is ") for the case where the duplicate keys lie in the middle of the file instead of the last.